backend/src/backend/utils.py

```diff
@@ -0,0 +1,28 @@
+from zimscraperlib.i18n import find_language_names
+
+
+def get_ident_from_name(name: str) -> str:
+    # Name could be either use mul (yes it's 3 letters)
+    # or the combination of language codes.
+    # return pattern should be {project-name}_{lang}_{classifier}
+
+    if "khan-academy" in name:
+        # input example:
+        # khan-academy-videos_ar_playlist-PLlLMVkrxknErJ44BvGjhtfYxU80a9q4ii
+        lang_codes = name.split("_")[1].split("-")
+        try:
+            for lang in lang_codes:
+                find_language_names(lang)
+            return f"khan-academy_{'-'.join(lang_codes)}_playlists"
+        except AttributeError:
+            return ""
+
+    else:
+        # this takes care of all the wiki sources
+        lang_codes = name.split("_")[1].split("-")
```

Comment:
This could fail is the format is not correct.

Reply:
@rgaudin how should I handle it? 

These are the only book names that didn't work with the method

```
tutorial-wikipedia
wikipedia_roa-tara_all
wikipedia_sh_top
wikistage_multi_all
wikipedia_be-tarask_all
wikipedia_sh_all
wikizimmer.pokepedia-pw_
kiwix.wikipedia_crh-latn_all
wikipedia_eml_all
```

Should I write a regex/pattern, to handle these cases as well?

Reply:
No, as discussed. First step is getting the 90% that fits. Leave those aside for now.

Reply:
Your function was assuming it could not fail and would return empty string on error. Users are allowed to send garbage to the endpoint and we have to make sure this doesn't crash and return the appropriate error code/response.

As mentioned elsewhere the failsafe approach is not what we want. In your code calling this function, you'll wrap it to catch exceptions.

Now here you can get an `IndexError` if the text is not composed of multiple `_`. We don't want our user to have to catch IndexError nor is it helpful to the enduser. Instead, you'll wrap this call in a try for `IndexError` and raise a `ValueError` mentionning the fact that the language code(s) could not be found.

Reply:
> the failsafe approach is not what we want

I'll keep this in mind